modules/userId/index.js

```diff
@@ -605,6 +605,57 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+ * This function will be exposed in global-name-space so that userIds for a source can be exposed
+ * Sample use case is exposing this function to ESP
+ */
+function getEncryptedEidsForSource(source, encrypt, customFunction) {
+  let eids = [];
+  let eidsSignals = {};
+
+  if (isFn(customFunction)) {
+    logInfo(`${MODULE_NAME} - Getting encrypted signal from custom function : ${customFunction.name} & source : ${source} `);
+    // Publishers are expected to define a common function which will be proxy for signal function.
+    let customSignals = customFunction(source);
+    eidsSignals[source] = customSignals ? encryptSignals(customSignals) : null; // by default encrypt using base64 to avoid JSON errors
+  } else {
+    // initialize signal with eids by default
+    eids = (getGlobal()).getUserIdsAsEids();
+    logInfo(`${MODULE_NAME} - Getting encrypted signal for eids :${JSON.stringify(eids)}`);
+    eids.forEach((eid) => {
+      eidsSignals[eid.source] = encrypt === true ? encryptSignals(eid) : eid.uids[0].id; // If encryption is enabled append version (1||) and encrypt entire object
+    });
+  }
+  let promise = Promise.resolve(eidsSignals[source]);
+  logInfo(`${MODULE_NAME} - Fetching encrypted eids: ${eidsSignals[source]}`);
+  return promise;
+}
+
+function encryptSignals(signals, version = 1) {
+  let encryptedSig = '';
+  switch (version) {
+    case 1: // Base64 Encryption
+      encryptedSig = typeof signals === 'object' ? window.btoa(JSON.stringify(signals)) : window.btoa(signals); // Test encryption. To be replaced with better algo
+      break;
+    default:
+      break;
+  }
+  return `${version}||${encryptedSig}`;
+}
+
+function registerSignalSources(gtag, signalSources, encrypt, customFunction) {
```

Comment:
@r-schweitzer Adding customFunction for each customSources can be done if the logic of fetching the data for each source is different. This can be done in near future.

Currently we have kept it as simple where custom sources data can be read from getcustomSignal function where all the data are kept under single entity for publisher to defined the first party data or segments on the page.

Reply:
@smenzer Added encryptedSignal object in userSync object which will be used
```
userSync: {
                ...,
                encryptedSignal: {
                    // Eids sources for which signals needs to registered.
                    "eids": {
                        "sources": [
                            "sharedid.org",
                            "criteo.com",
                            "id5-sync.com",
                            "pubcid.org",
                            "audigent.com"
                        ],
                        "encrypt": false
                    },
                    // Custom Sources having custom function which returns respective data
                    "custom": {
                        "sources": [{
                            source: ['pubmatic.com'],
                            customFunc: () => {
                                return '{"keywords":["tech","auto"]}';
                            },
                        }, {
                            source: ['segment.com'],
                            customFunc: () => {
                                return '[{ "id": "1", "value": "seg1" },{ "id": "2", "value": "seg2" }]';
                            },
                        }],
                        "encrypt": true // Encryption value which will encode the data in base64 format
                    },
                    "registerDelay": 3000 // To delay the Registration of function to create encrypted signals
                },
                ....
            }

```